src/System.Runtime/tests/System/Single.cs

```diff
@@ -176,10 +176,6 @@ public static void TestToString()
 
         Single i2 = -8249;
         Assert.Equal("-8249", i2.ToString());
-
-        Assert.Equal("NaN", Single.NaN.ToString());
-        Assert.Equal("Infinity", Single.PositiveInfinity.ToString());
```

Comment:
We could potentially roundtrip numbers through Single/Double.ToString("C", CultureInfo.CurrentCulture.NumberFormat).